L1Trigger/L1THGCal/src/backend/HGCalMulticlusteringHistoImpl.cc

```diff
@@ -177,19 +178,23 @@ HGCalMulticlusteringHistoImpl::Histogram HGCalMulticlusteringHistoImpl::fillSmoo
 
 
 
+std::vector<std::pair<GlobalPoint, double > > HGCalMulticlusteringHistoImpl::computeSecondaryMaxSeeds( const Histogram & histoClusters ){
 
-std::vector<GlobalPoint> HGCalMulticlusteringHistoImpl::computeMaxSeeds( const Histogram & histoClusters ){
+  std::vector<std::pair<GlobalPoint, double > > seedPositions;
 
-    std::vector<GlobalPoint> seedPositions;
+    std::unordered_map<int, std::unordered_map<int, std::unordered_map<int, bool> > > primarySeedPositions;
+    std::unordered_map<int, std::unordered_map<int, std::unordered_map<int, bool> > > secondarySeedPositions;
+    std::unordered_map<int, std::unordered_map<int, std::unordered_map<int, bool> > > vetoPositions;
```

Comment:
This data structure is overly complicated and this complication is not needed.
Use a single level of maps and a key that includes R, phi, and z